master/internal/logpattern/logpattern.go

```diff
@@ -61,9 +61,21 @@ func (l *logPatternPolicies) monitor(ctx context.Context,
 		if log.AgentID == nil {
 			return fmt.Errorf("agentID must be non nil to monitor logs")
 		}
+		// The first line of trial logs is printing expconf which has the regex pattern.
+		// We skip monitoring this line.
+		regex := "(.*)(\\\"log_pattern_policies\\\":)(.*)"
+		compiledRegex, err := l.getCompiledRegex(regex)
 
 		for _, policy := range policies {
-			compiledRegex, err := l.getCompiledRegex(policy.Pattern())
+			if err != nil {
+				return err
+			}
+			if compiledRegex.MatchString(log.Log) {
```

Comment:
I think this check can go outside of the policy loop

Reply:
Good catch!